code/modules/recycling/conveyor2.dm

```diff
@@ -138,24 +136,36 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	use_power(6)
 
 	//get the first 30 items in contents
-	affecting = list()
-	var/i = 0
-	var/list/items = loc.contents - src
-	for(var/item in items)
-		i++ // we're sure it's a real target to move at this point
-		if(i >= MAX_CONVEYOR_ITEMS_MOVE)
-			break
-		affecting.Add(item)
-
+	var/turf/locturf = loc
+	var/list/items = locturf.contents - src - locturf.lighting_object
+	if(!LAZYLEN(items))//Dont do anything at all if theres nothing there but the conveyor
+		return
+	var/list/affecting = list()
+	if(length(items) > MAX_CONVEYOR_ITEMS_MOVE)
+		affecting = items.Copy(1, MAX_CONVEYOR_ITEMS_MOVE + 1)//Lists start at 1 lol
+	else
+		affecting = items
 	conveying = TRUE
-	addtimer(CALLBACK(src, .proc/convey, affecting), 1)
+
+	addtimer(CALLBACK(src, .proc/convey, affecting), 1)//Movement effect
 
 /obj/machinery/conveyor/proc/convey(list/affecting)
-	for(var/atom/movable/A in affecting)
-		if(!QDELETED(A) && (A.loc == loc))
-			A.ConveyorMove(movedir)
-			//Give this a chance to yield if the server is busy
-			stoplag()
+	for(var/O in affecting)
+		if(!ismovable(O))	//This is like a third faster than for(var/atom/movable in affecting)
+			continue
+		var/atom/movable/A = O
+		//Give this a chance to yield if the server is busy
+		stoplag()
+		if(QDELETED(A) || (A.loc != loc))
+			continue
+		if(iseffect(A) || isdead(A))
+			continue
+		if(isliving(A))
+			var/mob/living/zoommob = A
+			if((zoommob.movement_type & FLYING) && !zoommob.stat)
+				continue
+			if(!A.anchored && A.has_gravity())
+				step(A, movedir)
```

Comment:
```suggestion
	for(var/am in affecting)
		if(!ismovable(am))	//This is like a third faster than for(var/atom/movable in affecting)
			continue
		var/atom/movable/movable_thing = am
		//Give this a chance to yield if the server is busy
		stoplag()
		if(QDELETED(movable_thing) || (movable_thing.loc != loc))
			continue
		if(iseffect(movable_thing) || isdead(movable_thing))
			continue
		if(isliving(movable_thing))
			var/mob/living/zoommob = movable_thing
			if((zoommob.movement_type & FLYING) && !zoommob.stat)
				continue
			if(!movable_thing.anchored && movable_thing.has_gravity())
				step(movable_thing, movedir)
```
Could we use `Move()` here instead of `step()` or would that cause troubles? If it did, we'd be able to respect grabs and custom blocks, for example.